go/consensus/tendermint/tendermint.go

```diff
@@ -934,7 +934,7 @@ func genesisToTendermint(d *genesisAPI.Document) (*tmtypes.GenesisDoc, error) {
 	}
 
 	doc := tmtypes.GenesisDoc{
-		ChainID:     d.ChainID,
+		ChainID:     d.ChainContext()[:tmtypes.MaxChainIDLen],
```

Comment:
So we always truncate here?

Reply:
Yeah, unfortunately Tendermint has `MaxChainIDLen` at `50`. We have a fork so we could change it :-)